packages/devtools_app/lib/src/screens/memory/panes/diff/controller/heap_diff.dart

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import '../../../shared/heap/heap.dart';
+import '../../../shared/heap/model.dart';
+
+/// Stores already calculated comparisons for heap couples.
+class HeapDiffStore {
+  final _store = <_HeapCouple, HeapComparison>{};
+
+  HeapComparison compare(AdaptedHeap heap1, AdaptedHeap heap2) {
+    final couple = _HeapCouple(heap1, heap2);
+    if (!_store.containsKey(couple)) _store[couple] = HeapComparison(couple);
+    return _store[couple]!;
```

Comment:
this can be replaced with:

`return _store.putIfAbsent(couple, () => HeapComparison(couple));`